### ISSUE TITLE
Discrepancy in experiment reproduction

### ISSUE THREAD

**Issue by palaskasx-star** [Hao840/manifold-distillation#2]
First of all, I found your work very inspiring and wanted to thank you for sharing your code.

I am opening this issue because I encountered a discrepancy when attempting to reproduce the training of a DeiT-Tiny student with a Cait-S24 teacher. I achieved an accuracy of 75.7% using the command line provided in your repository (see at the end), which differs from the 76.4% reported in your paper and repo.

To verify my setup, I also ran a baseline experiment using just the soft labels from the teacher with fixed depth, an experiment that is also examined in the paper. In that specific case, my training accuracy was 75.7% compared to your reported 75.8%. Since that gap is negligible, it suggests my setup is generally working correctly, but something specific to the main experiment configuration might be causing the performance drop.

I downloaded the teacher model from the link to the DeiT GitHub that is provided. My environment uses PyTorch==2.8.0 and timm==0.5.4.

Command line used from the repository:
python -m torch.distributed.launch --nproc_per_node=8 main.py --distributed --output_dir <output-dir> --data-path <dataset-dir> --teacher-path <path-of-teacher-checkpoint> --model deit_tiny_patch16_224 --teacher-model cait_s24_224 --distillation-type soft --distillation-alpha 1 --distillation-beta 1 --w-sample 0.1 --w-patch 4 --w-rand 0.2 --K 192 --s-id 0 1 2 3 8 9 10 11 --t-id 0 1 2 3 20 21 22 23 --drop-path 0
